File: ios/common/Networking/WebSocket/WalletTopUp.swift
//
//  WalletTopUp.swift
//  Shared
//
//  Created by Manly Man on 11/23/19.
//  Copyright © 2019 Innomalist. All rights reserved.
//

import Foundation

public final class WalletTopUp: SocketRequest {
    public typealias ResponseType = EmptyClass
    public var params: [Any]?
    
    public init(gatewayId: Int, currency: String, token: String, amount: Double) {
        self.params = [gatewayId, currency, token, amount]
    }
    
    required public init() {}
    
}

File: android/common/src/main/java/com/innomalist/taxi/common/networking/socket/WalletTopUp.kt
package com.innomalist.taxi.common.networking.socket

import com.innomalist.taxi.common.networking.socket.interfaces.SocketRequest

class WalletTopUp(gatewayId: Int, currency: String, token: String, amount: Double) : SocketRequest() {
    init {
        this.params = arrayOf(gatewayId, currency, token, amount)
    }
}